test/e2e/ec2/instance.go

```diff
@@ -163,6 +163,27 @@ func WaitForEC2InstanceRunning(ctx context.Context, ec2Client *ec2.Client, insta
 	return waiter.Wait(ctx, &ec2.DescribeInstancesInput{InstanceIds: []string{instanceID}}, nodeRunningTimeout)
 }
 
+func IsEC2InstanceImpaired(ctx context.Context, ec2Client *ec2.Client, instanceID string) (bool, error) {
+	describeStatusOutput, err := ec2Client.DescribeInstanceStatus(ctx, &ec2.DescribeInstanceStatusInput{
+		InstanceIds:         []string{instanceID},
+		IncludeAllInstances: aws.Bool(true),
+	})
+	if err != nil {
+		return false, fmt.Errorf("describing instance status %s: %w", instanceID, err)
+	}
+	for _, status := range describeStatusOutput.InstanceStatuses[0].SystemStatus.Details {
+		if status.Name == "reachability" {
```

Comment:
I'm curious why we are only checking for the name/type of status and not the status itself, like `passed` or `failed`? Also the `InstanceStatus` and `SystemStatus` response objects also have a `Status` field outside of the `Details` field, which provides the overall status of the instance or system health taking into account all the indivudual health reports in the `Details` field. This Status field takes the following values:`ok | impaired | insufficient-data | not-applicable | initializing`. Maybe we could just check for `ok` here instead of drilling into each of the reports details.

Reply:
Totally, we could. Each of the checks in instancesStatus and systemStatus roll up to that overall status you are referring to.

My thinking was to limit it down to what weve seen happen and we know is not due to our code. Then if new cases happen we can investigate them on a case by case, making sure we can confirm its really an ec2 thing that we should be retrying for.

Reply:
That makes sense but even in the limited check case, we need to check for `status.Name == "reachability"` and `status.Status == "failed"` right? The name check alone could lead to false positives, since the status could be `passed`.